detection_rules/rule.py

```diff
@@ -455,6 +459,23 @@ def process_note_plugins():
 
         return obj
 
+    @validates_schema
+    def validates_data(self, data, **kwargs):
+        """Validate rule directories against restricted directories."""
+        error_fields = [field for field in ['version', 'revision'] if data.get(field) is not None]
+        if not error_fields:
+            return
+
+        error_message = " and ".join(error_fields)
+
+        # Check each config directory against restricted directories
+        for config_dir in RULES_CONFIG.rule_dirs:
+            for restricted_dir in [RULES_DIR, RULES_BBR_DIR]:
+                if config_dir == restricted_dir or restricted_dir in config_dir.parents:
```

Comment:
We could just check for `os.environ["CUSTOM_RULES_DIR"]`, but it appears more obvious that any issues that may occur are due to the `RULES_CONFIG`.

Reply:
The problem with this approach is for users that choose to co-locate their custom rules within those directories. Which I suppose we can strongly advise against if needed.

Another option could be to look for `Elastic` in rule.author[0]

Reply:
oh that's a good idea. It would simplify this a lot.